src/components/AssetSearch/AssetList.tsx

```diff
@@ -23,17 +24,69 @@ type ItemData<T> = {
   handleClick: T
 }
 
+const sortAssetsByAccountAndMarketCap = (
+  assets: Asset[],
+  enrichedData: Record<string, { fiatAmount: string; cryptoAmount: string; marketCap: string }>,
+): Asset[] => {
+  return sortBy(assets, [
+    asset => {
+      return Number(enrichedData[asset.assetId].fiatAmount)
+    },
+    asset => {
+      return Number(enrichedData[asset.assetId].marketCap)
+    },
+  ]).reverse()
+}
+
+/**
+ * This function create a data object(cryptoAmount, fiatAmount, marketCap) from the Asset, portfolioAcountRows and marketData for each asset for facilitate sorting.
+ * @param assets
+ * @param portfolioAcountRows
+ * @param marketData
+ * @returns Record<string, { fiatAmount: string; cryptoAmount: string; marketCap: string }>
+ */
+export const createEnrichData = (
+  assets: Asset[],
+  portfolioAcountRows: AccountRowData[],
+  marketData: {
+    [x: string]: MarketData | undefined
+  },
+): Record<string, { fiatAmount: string; cryptoAmount: string; marketCap: string }> => {
+  const result = {} as Record<
```

Comment:
Such casting shouldn't be in app code. The only place where they should possibly be is if we're in the context of tests.